pkg/volume/azure_file/azure_provision.go

```diff
@@ -171,7 +173,12 @@ func (a *azureFileProvisioner) Provision(selectedNode *v1.Node, allowedTopologie
 		return nil, fmt.Errorf("claim.Spec.Selector is not supported for dynamic provisioning on Azure file")
 	}
 
-	account, key, err := a.azureProvider.CreateFileShare(name, account, sku, resourceGroup, location, requestGiB)
+	// when use azure file premium, account kind should be specified as FileStorage
+	accountKind := string(storage.StorageV2)
+	if strings.HasPrefix(strings.ToLower(sku), "premium") {
+		accountKind = string(storage.FileStorage)
+	}
```

Comment:
For the `azure premium file only supports pvc size >= 100GB` issue, could we add a check here? The error message `Cannot downgrade quota while the prior quota change is in progress` is misleading, we should replace it with a more clear message.

Reply:
@feiskyer we may not add a check here since `100GB` limit may change in the near future, there are also some constraints in quota size, it should be in a range, while we would not add chose checks either. I have already talked with azure premium file team, they think the error message is not correct, they would change that message.

Reply:
Looks reasonable. Please add the error message to release note, so that users could find it easily.

Reply:
After further investigation, I found this error message is caused by the second API call by set file share quota, I will try to address this issue in another PR: use one API call to create file share
https://github.com/kubernetes/kubernetes/blob/02cca1f11a8ae570369540564604d0cde06e0720/pkg/cloudprovider/providers/azure/azure_file.go#L70-L71

Reply:
@feiskyer I have fixed it in new commit, PTAL, now it's using one API call